src/test/java/hw5/tests/steps/ThenSteps.java

```diff
@@ -0,0 +1,101 @@
+package hw5.tests.steps;
+
+import hw5.JDIUser;
+import hw5.pages.DiffElemPage;
+import hw5.pages.UserTablePage;
+import hw5.utils.WebDriverSingleton;
+import io.cucumber.java.DataTableType;
+import io.cucumber.java.en.Then;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class ThenSteps extends AbstractStep {
+
+    public ThenSteps() throws IOException {
+        super();
+    }
+
+    @Then("There is an individual log row for each checkbox")
+    public void iAssertCheckboxLogsOnDifferentElementsPage(List<String> checkboxes) {
+        checkboxes = checkboxes.subList(1, checkboxes.size());
+        checkboxes.forEach(e -> assertTrue(diffElemPage.isCheckboxLogDisplayed(e, "true")));
+    }
+
+    @Then("There is a log row for radio button {string}")
+    public void iAssertRadioLogsOnDifferentElementsPage(String radioName) {
+        assertTrue(diffElemPage.isRadioLogDisplayed(radioName));
+    }
+
+    @Then("There is a log row for dropdown value {string}")
+    public void iAssertDropdownLogsOnDifferentElementsPage(String color) {
+        assertTrue(diffElemPage.isDropdownLogDisplayed(color));
+    }
+
+    @Then("\"User Table\" page should be opened")
+    public void iAssertTablePageTitle() {
+        assertEquals(WebDriverSingleton.getWebDriver().getTitle(), "User Table");
+    }
+
+    @Then("{int} Number Type Dropdowns should be displayed on Users Table on User Table Page")
+    public void iAssertDropdownsOnUserTablePage(int expectedDropdownsAmount) {
+        assertEquals(userTablePage.getUsersSelects().size(), expectedDropdownsAmount);
+        assertTrue(userTablePage.isUsersDropdownsDisplayed());
+    }
+
+    @Then("{int} Usernames should be displayed on Users Table on User Table Page")
+    public void iAssertUsernamesOnUserTablePage(int expectedUsernamesAmount) {
+        assertEquals(userTablePage.getUsersNames().size(), expectedUsernamesAmount);
+        assertTrue(userTablePage.isUsersNamesDisplayed());
+    }
+
+    @Then("{int} Description texts under images should be displayed on Users Table on User Table Page")
+    public void iAssertDescriptionsUnderImagesOnUserTablePage(int expectedDescriptionsAmount) {
+        assertEquals(userTablePage.getUsersImagesDescriptions().size(), expectedDescriptionsAmount);
+        assertTrue(userTablePage.isUsersImagesDescriptionDisplayed());
+    }
+
+    @Then("{int} checkboxes should be displayed on Users Table on User Table Page")
+    public void iAssertCheckboxesOnUserTablePage(int expectedCheckboxesAmount) {
+        assertEquals(userTablePage.getUsersCheckboxes().size(), 6);
+        assertTrue(userTablePage.isUsersCheckboxesDisplayed());
+    }
+
+    @DataTableType
+    public JDIUser JdiUserEntry(Map<String, String> entry) {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.